src/scripts/hubot-jenkins-notifier.coffee

```diff
@@ -117,22 +140,50 @@ class JenkinsNotifier
     if !data || typeof data.build != 'object'
       @error new Error("Unable to process data"), req.body
       return
+
+    @logMessage(query.trace, data.build)
+
     fullurl = data.build.full_url || data.build.url
 
+    if data.build.phase in ['COMPLETED']
+      @logMessage(query.trace, "Ignoring phase COMPLETED")
+      return
+
+    if data.build.phase in ['STARTED']
+      @logMessage(query.trace, "#{data.name} #{data.build.phase} #{data.build.status}")
+
+      @robot.send envelope, "#{data.name} build ##{data.build.number} started: #{fullurl}"
```

Comment:
Yes please

I'll be home Sunday and try to.give better feedback.

On Aug 19, 2016 1:49 PM, "Jens Doose" <notifications@github.com> wrote:

> In src/scripts/hubot-jenkins-notifier.coffee
> <https://github.com/halkeye/hubot-jenkins-notifier/pull/7#discussion_r75549311>
> :
>
> >      fullurl = data.build.full_url || data.build.url
> >
> > +    if data.build.phase in ['COMPLETED']
> > +      @logMessage(query.trace, "Ignoring phase COMPLETED")
> > +      return
> > +
> > +    if data.build.phase in ['STARTED']
> > +      @logMessage(query.trace, "#{data.name} #{data.build.phase} #{data.build.status}")
> > +
> > +      @robot.send envelope, "#{data.name} build ##{data.build.number} started: #{fullurl}"
>
> Hm, I am not so sure about this... As far as I understand shouldNotify is
> used to determine a state change. But for STARTED there is no state change,
> right? So I figured, if somebody turns on STARTED they probably want to be
> notified in any case.
> But if you want to we can still wrap it in shouldNotify
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/halkeye/hubot-jenkins-notifier/pull/7/files/a83dad2397b1f8afeee252d7df10018e88d38781#r75549311>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGuB5C6lK_Xk-B9T7MnC6AFViL7_bhwks5qhhbigaJpZM4JY0mC>
> .
>